docker/context/context.py

```diff
@@ -0,0 +1,190 @@
+import os
+import json
+from shutil import copyfile, rmtree
+from docker import utils
+from docker.constants import IS_WINDOWS_PLATFORM
+from docker.tls import TLSConfig
+from docker.context.config import get_meta_dir, get_meta_file, get_tls_dir
+
+
+class Context:
+    """A context."""
+    def __init__(self, name, orchestrator="swarm", endpoints=None):
+        if not name:
+            raise Exception("Name not provided")
+        self.name = name
+        self.orchestrator = orchestrator
+        self.endpoints = {
+            "docker": {
+                "Host": utils.parse_host(None, IS_WINDOWS_PLATFORM),
+                "SkipTLSVerify": False
+            }
+        } if not endpoints else endpoints
+        self.tls_cfg = {}
+        self.meta_path = "IN MEMORY"
+        self.tls_path = "IN MEMORY"
+
+    def set_endpoint(
+            self, name, host=None, tls_cfg=None, skip_tls_verify=False,
+            def_namespace=None):
+        if not name:
+            return
+        self.endpoints[name] = {
+            "Host": utils.parse_host(host, IS_WINDOWS_PLATFORM),
+            "SkipTLSVerify": skip_tls_verify
+        }
+        if def_namespace:
+            self.endpoints[name]["DefaultNamespace"] = def_namespace
+
+        if tls_cfg:
+            self.tls_cfg[name] = tls_cfg
+
+    def inspect(self):
+        return self.__call__()
+
+    @classmethod
+    def load_context(cls, name):
+        name, orchestrator, endpoints = Context.load_meta(name)
+        if name:
+            instance = cls(name, orchestrator, endpoints)
+            instance.load_certs()
+            instance.meta_path = get_meta_dir(name)
+            return instance
+        return None
+
+    @classmethod
+    def load_meta(cls, name):
+        # load metadata
+        metadata = {}
+        meta_file = get_meta_file(name)
+        if os.path.isfile(meta_file):
+            with open(meta_file) as f:
+                try:
+                    with open(meta_file) as f:
+                        metadata = json.load(f)
+                    for k, v in metadata["Endpoints"].items():
+                        metadata["Endpoints"][k]["SkipTLSVerify"] = bool(
+                            v["SkipTLSVerify"])
+                except (IOError, KeyError, ValueError) as e:
+                    # unknown format
+                    raise Exception("""Detected corrupted meta file for
+                        context {} : {}""".format(name, e))
+
+            return (
+                metadata["Name"], metadata["Metadata"]["StackOrchestrator"],
+                metadata["Endpoints"])
+        return None, None, None
+
+    def load_certs(self):
+        certs = {}
+        # set certs
+        tls_dir = get_tls_dir(self.name)
```

Comment:
```suggestion
        self.tls_path = get_tls_dir(self.name)
```

Then, you can remove line 101